mobile/screens/UploadVideoScreen.tsx

```diff
@@ -0,0 +1,159 @@
+import { StatusBar } from "expo-status-bar";
+import React, { useEffect, useState } from "react";
+import { Ionicons } from "@expo/vector-icons";
+import HeaderDate from "../components/HeaderDate";
+
+import {
+  BackHandler,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  Image,
+  Button,
+} from "react-native";
+import * as ImagePicker from "expo-image-picker";
+
+export default function AddTrainingLogScreen(props: any) {
+  const [image, setImage] = useState();
+
+  useEffect(() => {
+    BackHandler.addEventListener("hardwareBackPress", function () {
+      props.navigation.navigate("User Dashboard");
+      return true;
+    });
+  }, []);
+
+  const pickImage = async () => {
+    // No permissions request is necessary for launching the image library
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.Videos,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+    console.log(result);
+  };
+
+  return (
+    <View style={styles.container}>
+      <View>
+        <View style={styles.headerContainer}>
+          <TouchableOpacity
+            onPress={() => props.navigation.navigate("Add Training Log")}
+            style={styles.backArrow}
+          >
+            <Ionicons
+              name="chevron-back-circle"
+              size={45}
+              style={styles.arrowIcon}
+              color="grey"
+            />
+          </TouchableOpacity>
+          <View>
+            <HeaderDate />
+          </View>
+        </View>
+
+        <Text style={styles.label}>Video Log</Text>
+        <View style={styles.videoUploadContainer}>
+          <TouchableOpacity style={styles.uploadBtn} onPress={pickImage}>
+            <Ionicons name="add-circle-outline" size={30} color="grey" />
+            <Text style={styles.uploadText}> Training Video Log</Text>
+          </TouchableOpacity>
+          {image && <Image source={{ uri: image }} />}
+        </View>
+        <View style={styles.videoUploadContainer}>
+          <TouchableOpacity style={styles.uploadBtn} onPress={pickImage}>
+            <Ionicons name="add-circle-outline" size={30} color="grey" />
+            <Text style={styles.uploadText}> Training Video Log</Text>
+          </TouchableOpacity>
+          {image && <Image source={{ uri: image }} />}
+        </View>
+      </View>
+      <View>
+        <TouchableOpacity style={styles.button}>
+          <Text style={styles.buttonText}>Finish</Text>
+        </TouchableOpacity>
```

Comment:
when this is clicked, make sure you execute a method that calls `userCreateTrainingLog` from `mobile/actions/TrainingLog.ts`. This will require some information from the previous screen which I recommend you propagate over via navigation params (i.e. see how Nathan did it [here](https://github.com/GTBitsOfGood/Healing4Heroes/blob/76d40c6e0e1a7f6bf3dc69727dea546c6d6399a7/mobile/screens/HandlerInformationScreen.tsx#L118)